examples/pipeline/kg_builder.py

```diff
@@ -0,0 +1,91 @@
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from __future__ import annotations
+
+import asyncio
+from typing import Any
+
+from neo4j_genai.core.component import Component
+
+
+class DocumentChunker(Component):
+    async def run(self, text: str) -> dict[str, Any]:
+        return {"chunks": [t.strip() for t in text.split(".") if t.strip()]}
+
+
+class SchemaBuilder(Component):
+    async def run(self, schema: dict[str, Any]) -> dict[str, Any]:
+        return {"schema": schema}
+
+
+class ERExtractor(Component):
+    async def _process_chunk(self, chunk: str, schema: str) -> dict[str, Any]:
+        return {
+            "data": {
+                "entities": [{"label": "Person", "properties": {"name": "John Doe"}}],
+                "relations": [],
+            }
+        }
+
+    async def run(self, chunks: list[str], schema: str) -> dict[str, Any]:
+        tasks = [self._process_chunk(chunk, schema) for chunk in chunks]
+        result = await asyncio.gather(*tasks)
+        merged_result: dict[str, Any] = {"data": {"entities": [], "relations": []}}
+        for res in result:
+            merged_result["data"]["entities"] += res["data"]["entities"]
+            merged_result["data"]["relations"] += res["data"]["relations"]
+        return merged_result
+
+
+class Writer(Component):
+    async def run(
+        self, entities: dict[str, Any], relations: dict[str, Any]
+    ) -> dict[str, Any]:
+        return {
+            "status": "OK",
+            "entities": entities,
+            "relations": relations,
+        }
+
+
+if __name__ == "__main__":
+    from neo4j_genai.core.pipeline import Pipeline
+
+    pipe = Pipeline()
+    pipe.add_component("chunker", DocumentChunker())
+    pipe.add_component("schema", SchemaBuilder())
+    pipe.add_component("extractor", ERExtractor())
+    pipe.add_component("writer", Writer())
+    pipe.connect("chunker", "extractor", input_defs={"chunks": "chunker.chunks"})
+    pipe.connect("schema", "extractor", input_defs={"schema": "schema.schema"})
+    pipe.connect(
```

Comment:
Should there be some warning message or exception handling if the user does not connect the components correctly?

Reply:
Yes definitely. Can you help me in identifying the wrong ways? I can see the following for now:
1. Wrong component name (I think it's already covered, but will test)
2. Cyclic graph: this is already supported and will raise an exception
3. Missing input for a component: this is tricky, because some inputs can be provided in the `pipeline.run` stage, so they are unknown here (but maybe worth raising a warning anyway?)
4. Isolated component (Note: we will only know this when we start running the pipeline)

Also, I'd like to support some "automatic" propagation. Typically, when a component returns "text" and the next component is expecting a "text" input, then user do not have to explicitly define the mapping. I've let that for a later work, but I can rework on it if you think it is important.

Reply:
> Missing input for a component: this is tricky, because some inputs can be provided in the pipeline.run stage, so they are unknown here (but maybe worth raising a warning anyway?)

I think worth at least raising a warning asking along the lines of "Are all components connected as intended?". Earlier, I had forgot to add 

```python
pipe.connect(
        "extractor",
        "writer",
        input_defs={
            "entities": "extractor.data.entities",
            "relations": "extractor.data.relations",
        },
    )
```

and was confused when I got the error

```bash
TypeError: Writer.run() missing 2 required positional arguments: 'entities' and 'relations'
``

Reply:
Added a check before starting the pipeline (in the run method) here: https://github.com/neo4j/neo4j-genai-python/pull/81/commits/f0edf1562abc7aeb723ab93bfa6dda88ab3fcbbd

Reply:
Component output definition and checks added in https://github.com/neo4j/neo4j-genai-python/pull/81/commits/9c8bbd39a2bec379fe5f4bec938a9f837c41f129